release.py

```diff
@@ -0,0 +1,203 @@
+    put_message(os.path.join(PACKAGE_PATH, 'VERSION'), version)
+    build_messages_json(history)
+    commit_release(version)
+    print("Release %s created!" % version)
+
+
+def publish_release(token: str) -> None:
+    """Publish the new release."""
+    version = get_message(os.path.join(PACKAGE_PATH, 'VERSION'))
+
+    repo_url = 'https://github.com/{}'.format(GITHUB_REPO)
+    # push release branch to server
+    git('push', repo_url, RELEASE_BRANCH)
+    # push tags to server
+    git('push', repo_url, 'tag', version)
+
+    # publish the release
+    post_url = '/repos/{}/releases?access_token={}'.format(GITHUB_REPO, token)
+    headers = {
+        'User-Agent': 'Sublime Text',
+        'Content-type': 'application/json',
+    }
+    # get message from /messages/<version>.txt
+    text = get_message(os.path.join(MESSAGE_PATH, version + '.txt'))
+    # strip message header (version)
+    text = text[text.find('\n') + 1:]
+    # built the JSON request body
+    data = json.dumps({
+        "tag_name": version,
+        "target_commitish": RELEASE_BRANCH,
+        "name": version,
+        "body": text,
+        "draft": False,
+        "prerelease": False
+    })
+    try:
+        import http.client
+        client = http.client.HTTPSConnection('api.github.com')
+        client.request('POST', post_url, body=data, headers=headers)
+        response = client.getresponse()
+        print("Release %s published!" % version
+              if response.status == 201 else
+              "Release %s failed!" % version)
+    finally:
+        client.close()
+
+
+"""
+======================================
+Command Line Interface
+======================================
+"""
+if __name__ == '__main__':
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description='Built and Publish {} Releases'.format(__package__))
+    parser.add_argument(
+        dest='command',
+        help='The command to perform is one of [BUILD|PUBLISH].')
+    parser.add_argument(
+        '--token',
+        nargs='?',
+        help='The GitHub access token used for authentication.')
+    args = parser.parse_args()
+    if args.command.lower() == 'build':
+        build_release()
+    elif args.command.lower() == 'publish':
+        publish_release(args.token)
+
+
+"""
+======================================
+Sublime Text Command Interface
+======================================
+"""
+try:
+    import sublime
+    import sublime_plugin
+
+    camel_case_prefix = ''.join([word.title() for word in COMMAND_PREFIX.lower().split('_')])
+
+    class InternalBuildReleaseCommand(sublime_plugin.ApplicationCommand):
+
+        def is_visible(self) -> bool:
+            settings = sublime.load_settings(SETTINGS)
+            return settings.has('github_token')
+
+        def run(self) -> None:
+            """Built a new release."""
+            build_release()
+
+    InternalBuildReleaseCommand.__name__ = '{}BuildReleaseCommand'.format(camel_case_prefix)
```

Comment:
Why do it like this?

Reply:
For reusability. I'd like to use this script in LSP packages also and having to change this code for each of them is cumbersome. It's easier if all configuration is at the top.

To be fair, it would be even better if the script would read an external configuration file instead so it doesn't have to be adjusted at all.

Reply:
I see. I think a json configuration file would be better suited for this.

Reply:
Agreed, will do that.

Reply:
BTW. Alternatively, I could just remove those commands. The shell script should be enough really.

Reply:
I agree the shell scripts should be enough :)